src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java

```diff
@@ -303,33 +323,48 @@ private RevisionLocation zipAndUpload(AWSClients aws, String projectName, FilePa
 
             }
 
-            logger.println("Zipping files into " + zipFile.getAbsolutePath());
-
+            logger.println("package files into " + tarzipFile.getAbsolutePath());
 
-
-            sourceDirectory.zip(
-                    new FileOutputStream(zipFile),
+			if (fileType == null || fileType.equals("Zip")) {
```

Comment:
Might change this to switch based on bundleType since you set it earlier into 1 of the 3 categories (avoids having to have the fallback / null check here)